apps/telegram_bot/main.py

```diff
@@ -0,0 +1,335 @@
+from telegram import Update
+from telegram.ext import (
+    ApplicationBuilder,
+    ContextTypes,
+    CommandHandler,
+    MessageHandler,
+    filters,
+)
+
+from vocode.turn_based.transcriber.whisper_transcriber import WhisperTranscriber
+from vocode.turn_based.agent.chat_gpt_agent import ChatGPTAgent
+from vocode.turn_based.synthesizer.coqui_synthesizer import CoquiSynthesizer
+
+# Optional alternative synthesizers:
+from vocode.turn_based.synthesizer.stream_elements_synthesizer import (
+    StreamElementsSynthesizer,
+)
+from vocode.turn_based.synthesizer.eleven_labs_synthesizer import ElevenLabsSynthesizer
+from vocode.turn_based.synthesizer.play_ht_synthesizer import PlayHtSynthesizer
+from vocode.turn_based.synthesizer.azure_synthesizer import AzureSynthesizer
+from vocode.turn_based.synthesizer.coqui_tts_synthesizer import CoquiTTSSynthesizer
+from vocode.turn_based.synthesizer.rime_synthesizer import RimeSynthesizer
+
+from pydub import AudioSegment
+import io
+import logging
+import os
+import subprocess
+import inspect
+from typing import Tuple, Union
+import pickle
+
+logging.basicConfig(
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO
+)
+
+# Synthesizer class must be one of the following (do not change without adding support):
+supported_classes = [
+    CoquiSynthesizer,
+    StreamElementsSynthesizer,
+    ElevenLabsSynthesizer,
+    PlayHtSynthesizer,
+    AzureSynthesizer,
+    CoquiTTSSynthesizer,
+    RimeSynthesizer,
+]
+""" CONFIGURATION START """
+# Required environment variables containing API key: OPENAI_API_KEY, TELEGRAM_BOT_KEY, and your Vocode synthesizers classes corresponding API key variable
+
+# Your chosen synthesizer provider's corresponding Vocode turn_based class instance
+SYNTH = CoquiSynthesizer()
+
+# Array of tuples (synthesizer's voice id, nickname, description if text to voice)
+DEFAULT_VOICES = [("d2bd7ccb-1b65-4005-9578-32c4e02d8ddf", "Coqui Default", None)]
+
+# instructions to the language model responsible for generating response transcript.
+SYSTEM_PROMPT = "You are having a voice-to-voice conversation with a user. Always pretend you are a real human. Do not avoid or decline to answer questions because you are not human."
+""" CONFIGURATION END """
+
+# key=synth class, value=attribute that chooses the provider voice
+voice_attr_of = {
+    ElevenLabsSynthesizer: "voice_id",
+    PlayHtSynthesizer: "voice",
+    StreamElementsSynthesizer: "voice",
+    AzureSynthesizer: "voice_name",
+    CoquiSynthesizer: "voice_id",
+    CoquiTTSSynthesizer: "speaker",
+    RimeSynthesizer: "speaker",
+}
+assert set(voice_attr_of.keys()) == set(
```

Comment:
we can remove these asserts

Reply:
If I don't have these I get type errors.